velox/vector/VectorSaver.cpp

```diff
@@ -508,6 +513,56 @@ VectorPtr readMapVector(
   return std::make_shared<MapVector>(
       pool, type, nulls, size, offsets, sizes, keys, values);
 }
+
+void writeLazyVector(const BaseVector& vector, std::ostream& out) {
+  auto lazyVector = dynamic_cast<const LazyVector*>(&vector);
+  // Nulls buffer.
+  bool isLoaded = lazyVector->isLoaded();
+  write<bool>(isLoaded, out);
+
+  // loaded vector.
+  if (isLoaded) {
```

Comment:
What happens if the vector is not loaded? Looks like we are not persisting anything in this case. How will it be restored?

Reply:
The restored lazy vector will only have the format and size and would fail if it an attempt is made to load it. the idea here is to be able to repro an issue, so if the repro happened before the lazy vector got materialized, this will let us create a placeholder for it with the right format instead of skipping vector serialization altogether and allow for a quick repro.

Reply:
@bikramSingh91 Thank you for explaining. Makes sense. It would be great to document this as it is not intuitive.